src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,231 @@
+
+    attr_accessor :context
+
+    def initialize(snapshot = nil)
+      @context = snapshot
+    end
+
+    def all(configs)
+      configs ||= list_configs
+      dbus_snapshots = Array(configs).each_with_object([]) do |config, snapshot_list|
+        list_snapshots(config).each do |dbus_snapshot|
+          next if dbus_snapshot[0] == 0
+          snapshot_list << {
+            type:        SNAPSHOT_ID[dbus_snapshot[1]],
+            number:      dbus_snapshot[0],
+            pre_number:  dbus_snapshot[2].to_i,
+            timestamp:   Time.at(dbus_snapshot[3]),
+            uid:         dbus_snapshot[4],
+            description: unescape(dbus_snapshot[5]),
+            cleanup:     unescape(dbus_snapshot[6]),
+            user_data:   unescape(dbus_snapshot[7]),
+            config:      config
+          }
+        end
+      end
+
+      dbus_snapshots
+    end
+
+    def create
+      attrs = [config || "", pre_number, description, cleanup, user_data]
+
+      attrs.delete_at(1) if !@context.post?
+      log.info "Creating snapshot #{@context.inspect} with result #{attrs.inspect}."
+
+      result = SnapshotDBus.dbus_object.send("Create#{@context.name.capitalize}Snapshot", *attrs).first
+
+      log.info "Creating snapshot #{@context.inspect} with result #{result}."
+
+      result
+    end
+
+    def description
+      escape(@context.description || "")
+    end
+
+    def cleanup
+      escape(@context.cleanup || "")
+    end
+
+    def user_data
+      escape(@context.user_data || {})
+    end
+
+    def update
+      result = SnapshotDBus.dbus_object.SetSnapshot(config, number, description, cleanup, user_data).first
+
+      log.info "Updating snapshot #{@context.inspect} with result #{result.inspect}."
+
+      result
+    end
+
+    def delete
+      result = SnapshotDBus.dbus_object.DeleteSnapshots(config, [number]).first
+      log.debug("delete_snapshots config:#{config} nums:#{number} result:#{result}")
+
+      result
+    end
+
+    def mount_point
+      result = SnapshotDBus.dbus_object.GetMountPoint(config, number).first
+      log.debug("Mount point for config_name:#{config} num:#{number} result#{result}")
+
+      result
+    end
+
+    def list_configs
+      result = SnapshotDBus.dbus_object.ListConfigs().first
+      log.debug("list_configs result:#{result}")
+
+      result.map(&:first)
+    end
+
+    def get_config(config_name)
+      result = SnapshotDBus.dbus_object.GetConfig(config_name).first
+      log.debug("get_config for name '#{config_name}' result:#{result}")
+
+      result
+    end
+
+    def list_snapshots(config_name = nil)
+      result = SnapshotDBus.dbus_object.ListSnapshots(config_name).first
+      log.debug("list_snapshots for name #{config_name} result:#{result}")
+      result
+    end
+
+    def get_modified_files(from, to)
+      create_comparison(from.config, from, to)
+      files = get_files(from.config, from, to)
+      delete_comparison(from.config, from, to)
```

Comment:
I suggest to use for getting comparison block form, otherwise not much safe when exception raised